src/compiler/types.ts

```diff
@@ -293,6 +293,8 @@ module ts {
         Const =             0x00001000,  // Variable declaration
         OctalLiteral =      0x00002000,
 
+        SimpleArrowFunction =      0x00004000,  // Arrow function without parenthesized parameters

```

Comment:
Do we really need this flag? Are there any syntactic ambiguities we can get into?

I'd say that if an arrow function
  * has a single parameter and
  * that parameter has no default initializer and
  * that parameter is not a rest parameter

then emit it as a simple arrow function.